src/hacbs/components/ApplicationDetails/tabs/overview/visualization/utils/node-utils.ts

```diff
@@ -0,0 +1,265 @@
+const TESTS_DESC = `An integration test is a test pipeline you build and add to the application CI/CD definition. It tests each component as it’s added and then tests the entire application as a whole.`;
+const STATIC_ENV_DESC = `A static environment is a set of compute resources bundled together. Use static environments for developing, testing, and staging before releasing your application. You can share static environments across all applications within the workspace.`;
+const RELEASE_DESC = `After pushing your application to release, your application goes through a series of tests through the release pipeline to ensure the application complies with the  release policy set on the release target, also known as the "managed environment".`;
+const MANAGED_ENV_DESC = `A managed environment is your application release target. This release target is an external environment, set up in an external workspace, and managed by another team. It is set for each application and not automatically shared among applications within the workspace.`;
+
+export const TYPE_DESCRIPTIONS = {
+  [WorkflowNodeType.COMPONENT]: COMPONENT_DESC,
+  [WorkflowNodeType.BUILD]: BUILD_DESC,
+  [WorkflowNodeType.TESTS]: TESTS_DESC,
+  [WorkflowNodeType.COMPONENT_TEST]: TESTS_DESC,
+  [WorkflowNodeType.APPLICATION_TEST]: TESTS_DESC,
+  [WorkflowNodeType.STATIC_ENVIRONMENT]: STATIC_ENV_DESC,
+  [WorkflowNodeType.RELEASE]: RELEASE_DESC,
+  [WorkflowNodeType.MANAGED_ENVIRONMENT]: MANAGED_ENV_DESC,
+};
+
+export const statusToRunStatus = (status: string): RunStatus => {
+  switch (status) {
+    case runStatus.Succeeded:
+    case GitOpsDeploymentHealthStatus.Healthy:
+      return RunStatus.Succeeded;
+    case runStatus.Failed:
+    case runStatus.FailedToStart:
+    case GitOpsDeploymentHealthStatus.Degraded:
+      return RunStatus.Failed;
+    case runStatus.Running:
+    case GitOpsDeploymentHealthStatus.Progressing:
+    case runStatus['In Progress']:
+      return RunStatus.Running;
+    case 'PR needs merge':
+      return RunStatus.Cancelled; // to show a warning
+    case RunStatus.Pending:
+    case GitOpsDeploymentHealthStatus.Suspended:
+    case GitOpsDeploymentHealthStatus.Missing:
+    case GitOpsDeploymentHealthStatus.Unknown:
+      return RunStatus.Pending;
+    default:
+      return undefined;
+  }
+};
+
+export const getLinkForElement = (
+  element: Node<PipelineNodeModel, WorkflowNodeModelData>,
+): { tab: string; filter?: { name: string; value: string } } => {
+  const { workflowType, label, isDisabled, resources, groupNode, status } = element.getData();
+
+  switch (workflowType) {
+    case WorkflowNodeType.COMPONENT:
+      return {
+        tab: 'components',
+        filter: !groupNode && !isDisabled ? { name: 'name', value: label } : undefined,
+      };
+    case WorkflowNodeType.BUILD:
+      if (BUILD_PIPELINE_STATUSES.includes(status)) {
```

Comment:
There can be more statuses like  `Idle`, `Skipped`, `Cancelled` and `Failed`,  so instead of having this whitelist of statuses, can we have the check for empty state instead?